pkg/controller/signup_test.go

```diff
@@ -99,32 +95,24 @@ func (s *TestSignupSuite) TestSignupPostHandler() {
 		require.NoError(s.T(), err)
 		expectedUserID := ob.String()
 		ctx.Set(context.SubKey, expectedUserID)
+		ctx.Set(context.UsernameKey, "bill")
 		ctx.Set(context.EmailKey, expectedUserID+"@test.com")
-		signup := &crtapi.UserSignup{
-			TypeMeta: v1.TypeMeta{},
-			ObjectMeta: v1.ObjectMeta{
-				Name:      userID.String(),
-				Namespace: "namespace-foo",
-			},
-			Spec: crtapi.UserSignupSpec{
-				IdentityClaims: crtapi.IdentityClaimsEmbedded{
-					PreferredUsername: "bill",
-				},
-			},
-			Status: crtapi.UserSignupStatus{
-				Conditions: []crtapi.Condition{
+		signup := testusersignup.NewUserSignup(
+			testusersignup.WithName("bill"),
+			func(userSignup *crtapi.UserSignup) {
```

Comment:
`testusersignup.WithIncompleteSignup("test_reason", "test_message")` would make the test more readable IMHO, even if it this would be the single use of it.

Reply:
done in a103f78